### ISSUE TITLE
1.4.3 fails to compile with ocaml-mm 0.6.0

### ISSUE THREAD

**Issue by paddatrapper** [savonet/liquidsoap#1395]
I think this is because of the new upload of ocaml-mm 0.6.0 to Debian, but compiling liquidsoap 1.4.3 results in

```
OCAMLOPT -c stream/frame.ml
File "stream/frame.ml", line 363, characters 25-59:
363 |     contents = [(!!size, create_content (type_of_kind kind))];
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: This expression has type
         (float array array, Video.t array, MIDI.buffer array) fields
       but an expression was expected of type
         content = (audio_t array, video_t array, midi_t array) fields
       Type float array is not compatible with type
         audio_t =
           (float, Bigarray.float32_elt, Bigarray.c_layout) Bigarray.Array1.t 
make[4]: *** [../Makefile.rules:125: stream/frame.cmx] Error 2
```

I think I am going to hold the rest of Liquidsoap's dependencies updates out of unstable until after the next Debian release (freeze in January) or liquidsoap 2.0 is out

**Comment by MattMski** [comment]
I had a similar issue updating Debian tonight. I updated my ocaml from 4.09 to the latest 4.11.1 and that fixed my problem for upgrading.

**Comment by paddatrapper** [comment]
I still get it building Liquidsoap 1.4.3 with Ocaml 4.11.1

**Comment by toots** [comment]
Hi all! Yes, indeed. Since the current development cycle is expected to take time and involved breaking APIs from our underlying modules, we have started releasing versions of those modules that are targeted for the upcoming version to make switching back and forth easier with `opam`. However, these versions should be carefully considered when updating a distribution's package.

I believe that, for most of these packages, we are the only user so it's probably best to just wait until the next major release to publish them. If you really need to have the new version, I'm not actually sure how this could be done. I don't think that OCaml provides for multiple versions of the same module installed concurrently. You probably would have to put restrictions/conflicts on the module's package itself.

For a reliable source of which package to use for which version, I'd suggest to look at the `opam` files for `1.4.3` here: https://github.com/ocaml/opam-repository/blob/master/packages/liquidsoap/liquidsoap.1.4.3/opam and the `opam` file for the latest development version here: https://github.com/savonet/liquidsoap/blob/master/liquidsoap.opam

**Comment by toots** [comment]
@paddatrapper For debian specifically, if you need to rollback the upload of `ocaml-duppy`, I believe package epoch would actually be a good fit for that..

**Comment by paddatrapper** [comment]
@toots, duppy hasn't been updated since before Buster was released 2 years ago (version 0.8.0-1). Do you mean `ocaml-mm`?

**Comment by toots** [comment]
Yes sorry I mean `mm` :-)
